src/client/js/search-utilities.js

```diff
@@ -0,0 +1,48 @@
+/**
+*
+*  Online store PWA sample.
+*  Copyright 2017 Google Inc. All rights reserved.
+*
+*  Licensed under the Apache License, Version 2.0 (the "License");
+*  you may not use this file except in compliance with the License.
+*  You may obtain a copy of the License at
+*
+*      https://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing, software
+*  distributed under the License is distributed on an "AS IS" BASIS,
+*  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*  See the License for the specific language governing permissions and
+*  limitations under the License
+*
+*/
+
+const searchInput = document.querySelector('#search input');
+const searchSvg = document.querySelector('#search svg');
+
+const MIN_QUERY_LENGTH = 3;
+
+class SearchUtilities {
+  init() {
```

Comment:
Do you plan to have more methods here? Otherwise, maybe just a plain function (without a class) would do?

Reply:
My general feeling is that we should follow the same format for all 'helpers' like this: a class with an `init()` method (i.e. avoiding side effects on instantiation) and/or with other public methods. What do you think?

I also think that it's quite likely that more methods would be added later.

@rowan-m fyi

Reply:
I'm usually in favour of keeping things simple wherever possible. There is an overhead to using a class - one is that some people are still unfamiliar with it, second one is that there is a performance overhead we introduce when creating a class (for this particular example check here:  https://jsperf.com/pieshop-class-vs-function). So I'd use a class only if there is a need for one (e.g. private methods/variables, public api). I'm not sure if just sticking to the format ticks this box, but can be convinced otherwise - @rowan-m wdyt?

Reply:
Putting some of the thoughts from discussion this morning in here. This feels like a place where there's a specific element on the page and this is the JavaScript behaviour linked to it. We should define a convention to follow here. Is the [MDC web pattern](https://material.io/develop/web/docs/getting-started/) a good one? i.e. we end up with something like:
```html
<input type="search" class="pie-search" …>
```

```js
pie.SearchInput.attachTo(document.querySelector('.pie-search'));
```

Alternatively, are web components a good choice here instead?

Reply:
@devnook @rowan-m 

Excellent idea to test performance, though I think in this case (particularly because it's a singleton), this isn't a problem. The difference in performance is around 0.05 ms.

We could standardise in a couple of ways:
• Always use a class, but with no side effects on instantiation. Always use an `init()` or (maybe better) one or more methods (e.g. .
• Classes only where that makes sense, i.e. there is more than one function

I agree with Rowan that search-utilities is a bad name :). I think I'll rename this search-input.js. WDYT?

We also talked about using components (of some kind) but that's a whole 'nother discussion...